diff --git a/plugin/build/ios/appDelegate.js b/plugin/build/ios/appDelegate.js
index c1e95803368b6b6bb1ddf9cdcbdade2a8a91a664..68bc955660961f0d3b833a5f7ff5b160568c3d7d 100644
--- a/plugin/build/ios/appDelegate.js
+++ b/plugin/build/ios/appDelegate.js
@@ -61,7 +61,7 @@ function modifyObjcAppDelegate(contents) {
 }
 function modifySwiftAppDelegate(contents) {
     const methodInvocationBlock = `FirebaseApp.configure()`;
-    const methodInvocationLineMatcher = /(?:self\.moduleName\s*=\s*"([^"]*)")/g;
+    const methodInvocationLineMatcher = /(?:self\.moduleName\s*=\s*"([^"]*)")|(?:factory\.startReactNative\()/;
     // Add import
     if (!contents.includes('import FirebaseCore')) {
         contents = contents.replace(/import Expo/g, `import Expo
diff --git a/plugin/src/ios/appDelegate.ts b/plugin/src/ios/appDelegate.ts
index e611cd3e28832349a034a74df1728263a1797efb..b87fd7a3e3bfd4f7635758f409fdfc25413ac7c6 100644
--- a/plugin/src/ios/appDelegate.ts
+++ b/plugin/src/ios/appDelegate.ts
@@ -70,7 +70,7 @@ export function modifyObjcAppDelegate(contents: string): string {
 
 export function modifySwiftAppDelegate(contents: string): string {
   const methodInvocationBlock = `FirebaseApp.configure()`;
-  const methodInvocationLineMatcher = /(?:self\.moduleName\s*=\s*"([^"]*)")/g;
+  const methodInvocationLineMatcher = /(?:self\.moduleName\s*=\s*"([^"]*)")|(?:factory\.startReactNative\()/;
 
   // Add import
   if (!contents.includes('import FirebaseCore')) {
